docs/development_guide/README.md

```diff
@@ -164,6 +184,48 @@ version installed, run `go install -mod=readonly golang.org/x/tools/cmd/goimport
 other development tools to use `goimports`. Formatting is checked during CI by
 the `bin/fmt` script.
 
+## Putting it all together (inner development loop)
+
+Now that you have an overview of the various parts of the project such as build commands, environment variables and linting, we will give you a example 
+workflow for development.  Modify as need for your environment if using kind locally doesn't fit your use case.
+
+```bash
+# use default setup
+cp .env.example .env
+source .env
+
+# create a local kind cluster
+make kind-up
+
+# build osm control plane components and cli (includes changes to helm chart)
+make build-osm-all
+
+# deploy osm
+./bin/osm install --set=osm.image.registry="$CTR_REGISTRY" --set=osm.image.pullPolicy=Always --set=osm.controllerLogLevel=trace --verbose 
+```
+
+> **note**: you can set any of the OSM chart parameters in [values.yaml](/charts/osm/values.yaml) such as osm.image.registry on OSM install using `--set`. For example, if you need to customize the deployment tags you can use `--set=osm.image.registry="$CTR_TAG"`
+
+### Making changes to OSM 
+Make required changes to the OSM controller and deploy them.  You could do this for any server component, see more on the [Makefile](#makefile) for how to build individual containers.
+
+```bash
+# only builds controller (see below for all build commands)
+make docker-build-osm-controller
+kubectl rollout restart deployment osm-controller  
+```
+
+If you made changes to OSM cli or helm chart you might want to refresh your deployment:
+
+```bash
+# clean up and rebuild everything
+bin/osm uninstall mesh -f --mesh-name "$MESH_NAME" --delete-namespace -a
+make build-osm-all
```

Comment:
nit: include `--delete-cluster-wide-resources` to clean up everything as the comment suggests

Reply:
I think `-a` does this:

```
osm uninstall -h


Flags:
      --ca-bundle-secret-name string    Name of the secret for the OSM CA bundle (default "osm-ca-bundle")
  -a, --delete-cluster-wide-resources   Cluster wide resources (such as osm CRDs, mutating webhook configurations, validating webhook configurations and osm secrets) are fully deleted from the cluster after control plane components are deleted.

```

Reply:
Should probably just drop `-a` in favor of `--all` in the CLI.